pkg/apis/core/validation/shoot.go

```diff
@@ -556,6 +557,62 @@ func ValidateTotalNodeCountWithPodCIDR(shoot *core.Shoot) field.ErrorList {
 	return allErrs
 }
 
+func validateKubeAPIServerUpdate(newShoot, oldShoot *core.Shoot, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs               = field.ErrorList{}
+		oldEncryptedResources = sets.New[string]()
+		newEncryptedResources = sets.New[string]()
+		oldExcludedResources  = sets.New[string]()
+		newExcludedResources  = sets.New[string]()
+	)
+
+	if oldKubeAPIServer := oldShoot.Spec.Kubernetes.KubeAPIServer; oldKubeAPIServer != nil && oldKubeAPIServer.EncryptionConfig != nil {
+		oldEncryptedResources.Insert(oldKubeAPIServer.EncryptionConfig.Resources...)
+		oldExcludedResources.Insert(oldKubeAPIServer.EncryptionConfig.ExcludedResources...)
+	}
+
+	if newKubeAPIServer := newShoot.Spec.Kubernetes.KubeAPIServer; newKubeAPIServer != nil && newKubeAPIServer.EncryptionConfig != nil {
+		newEncryptedResources.Insert(newKubeAPIServer.EncryptionConfig.Resources...)
+		newExcludedResources.Insert(newKubeAPIServer.EncryptionConfig.ExcludedResources...)
+
+		for i, resource := range newKubeAPIServer.EncryptionConfig.ExcludedResources {
+			if oldExcludedResources.Has(resource) {
+				continue
+			}
+
+			if strings.HasPrefix(resource, "*.") && oldEncryptedResources.Has("*.*") {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("encryptionConfig", "excludedResources").Index(i), "cannot exclude this resource since encryptionConfig.resources contains a wildcard item for all resource groups. ie;'*.*' which means the resource is already encrypted"))
```

Comment:
```suggestion
				allErrs = append(allErrs, field.Forbidden(fldPath.Child("encryptionConfig", "excludedResources").Index(i), "cannot exclude this resource since encryptionConfig.resources contains a wildcard item for all resource groups, i.e., '*.*' which means the resource is already encrypted"))
```

same for other occurrences